pkg/log/log.go

```diff
@@ -0,0 +1,235 @@
+package log
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/codeready-toolchain/registration-service/pkg/context"
+	"github.com/operator-framework/operator-sdk/pkg/log/zap"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+
+	"github.com/gin-gonic/gin"
+	"github.com/go-logr/logr"
+	sync "github.com/matryer/resync"
+	"github.com/spf13/pflag"
+)
+
+var (
+	logger *Logger
+	once   sync.Once
+)
+
+type Logger struct {
+	logr logr.Logger
+}
+
+// Init initializes the logger.
+func Init(name string, out io.Writer, development bool) {
+	zapFlagSet := pflag.NewFlagSet("zap", pflag.ExitOnError)
+
+	// Add the zap logger flag set to the CLI. The flag set must
+	// be added before calling pflag.Parse().
+	pflag.CommandLine.AddFlagSet(zapFlagSet)
+
+	// Add flags registered by imported packages (e.g. glog and
+	// controller-runtime)
+	pflag.CommandLine.AddGoFlagSet(flag.CommandLine)
+
+	pflag.Parse()
+
+	// Use a zap logr.Logger implementation. If none of the zap
+	// flags are configured (or if the zap flag set is not being
+	// used), this defaults to a production zap logger.
+	//
+	// The logger instantiated here can be changed to any logger
+	// implementing the logr.Logger interface. This logger will
+	// be propagated through the whole operator, generating
+	// uniform and structured logs.
+	logf.SetLogger(zap.Logger())
```

Comment:
Why not moving everything to `once.Do(func() {...})`?

Reply:
Done!